src/core/distributions/phylogenetics/tree/TopologyConstrainedTreeDistribution.cpp

```diff
@@ -54,14 +53,16 @@ TopologyConstrainedTreeDistribution::TopologyConstrainedTreeDistribution(TypedDi
     dirty_nodes( base_dist->getValue().getNumberOfNodes(), true ),
     monophyly_constraints( c ),
     num_backbones( 0 ),
-    use_multiple_backbones( false )
-
+    use_multiple_backbones( false ),
+    starting_tree( t ),
+    rooting_known( false ),
+    is_rooted( true )
 {
-    AbstractRootedTreeDistribution* tree_base_distribution = dynamic_cast<AbstractRootedTreeDistribution*>(base_distribution);
-    if (tree_base_distribution == NULL)
-    {
-        throw(RbException("Can only constrain tree distributions of type AbstractRootedTreeDistribution"));
-    }
+//    AbstractRootedTreeDistribution* tree_base_distribution = dynamic_cast<AbstractRootedTreeDistribution*>(base_distribution);
```

Comment:
Why is this commented out?

Reply:
because I added code to work on unrooted tree too. I removed the check.